### PR TITLE
Theo/issue spec tests

### PR DESCRIPTION
**Pull request by theodorebugnet**
Added pre/post condition based tests to issue for request, execute and cancel.

Some existing tests were folded into the condition tests. All the relevant tests (old, and all the new ones) are organised into `mod_request_tests`, `mod_execute_tests` and `mod_cancel_tests`.

### CODE REVIEW

File: standalone/runtime/tests/test_issue.rs [hunk 1]
```diff
@@ -494,182 +572,411 @@ mod execute_refund_payment_limits {
...
+            drop_exchange_rate_and_liquidate(VAULT);
+            let post_liquidation_status = ParachainState::get();
+
+            execute_issue(issue_id);
+
+            assert_eq!(
+                ParachainState::get(),
+                post_liquidation_status.with_changes(|user, _vault, liquidation_vault, fee_pool| {
+                    user.free_tokens += issue.amount;
+                    // fee_pool.vault_rewards += issue.fee; // this is not the case - verify that's
```
Comment (theodorebugnet) [inline standalone/runtime/tests/test_issue.rs:870]: I'm not sure about this - the code seems to add vault rewards unconditionally: https://github.com/interlay/interbtc/blob/master/crates/issue/src/lib.rs#L445
However, the test fails if this assertion is enabled.

File: standalone/runtime/tests/mock/issue_testing_utils.rs [hunk 2]
```diff
@@ -15,6 +15,7 @@ pub struct RequestIssueBuilder {
     amount_btc: u128,
     vault: [u8; 32],
     user: [u8; 32],
+    collateral: u128,
```
Comment (gregdhill) [inline standalone/runtime/tests/mock/issue_testing_utils.rs:18]: Rename to `griefing_collateral` or so to be explicit.

File: standalone/runtime/tests/test_issue.rs (hunk 1 above)
Comment (gregdhill) [reply standalone/runtime/tests/test_issue.rs:870]: Ah yes, the problem here is that if there is only one Vault and if it is liquidated all rewards are sent to the treasury.

**Review by gregdhill — COMMENTED** [review]
I think it might be worth adding some more comments for which conditions we are specifically testing, otherwise this looks fine to me.

**Review by gregdhill — APPROVED** [review]

File: standalone/runtime/tests/test_issue.rs [hunk 3]
```diff
@@ -494,182 +584,442 @@ mod execute_refund_payment_limits {
...
+            );
+
+            // issue request status is set to cancelled
+            let user_issues = IssuePallet::get_issue_requests_for_account(account_of(USER));
+            let (_, onchain_issue) = user_issues.iter().find(|(id, _)| id == &issue_id).unwrap();
+            assert_eq!(onchain_issue.status, IssueRequestStatus::Cancelled);
+        });
+    }
+
+    /// Test Cancle preconditions in the case that the vault was liquidated
```
Comment (gregdhill) [inline standalone/runtime/tests/test_issue.rs:992]:
```suggestion
    /// Test cancel preconditions in the case that the vault was liquidated
```

**Review by gregdhill — CHANGES_REQUESTED** [review]

**Review by gregdhill — APPROVED** [review]

**Pull request merged**